src/luarocks/fs/lua.lua

```diff
@@ -290,7 +289,6 @@ function fs_lua.copy(src, dest, perms)
    if destmode == "directory" then
       dest = dir.path(dest, dir.base_name(src))
    end
-   if not perms then perms = fs.attributes(src, "permissions") end
```

Comment:
(Edited a comment I wrote earlier, which was wrong)

This is tricky: this needs to come back. We need `fs.attributes` to get the exact permissions when copying the files over. We could do this using `lfs.attributes` here (which is portable for Windows and Linux) but to _set_ the attributes we need Unix-specific code (on the other hand, we only really need to worry about these permissions on Unix, I think).

My current suggestion is this:

```lua
local fullattrs
if not perms then
   fullattrs = lfs.attributes(src, "permissions")
end
```

and then below for the second part...